lib/player.js

```diff
@@ -1254,11 +1256,15 @@ shaka.Player = class extends shaka.util.FakeEventTarget {
       mimeType = shaka.Player.SRC_EQUAL_EXTENSIONS_TO_MIME_TYPES_[extension];
     }
 
-    // TODO: The load graph system has a design limitation that requires routing
-    // destination to be chosen synchronously.  This means we can only make the
-    // right choice about src= consistently if we have a well-known file
-    // extension or API-provided MIME type.  Detection of MIME type from a HEAD
-    // request (as is done for manifest types) can't be done yet.
+    // If no MIME type is provided, and we can't base it on extension, make a
+    // HEAD request to determine it.
+    if (!mimeType) {
+      // TODO: Modify |HlsParser.guessMimeType_| to also use
+      // |shaka.media.ManifestParser.getMimeType|, for standardization?
```

Comment:
Maybe standardization should be to create net_utils.js with the most common functions.

Reply:
@theodab If you want, I can do it in a separate PR.

Reply:
https://github.com/shaka-project/shaka-player/pull/5756

Reply:
Thanks.